numpy/core/setup.py

```diff
@@ -685,13 +685,18 @@ def get_mathlib_info(*args):
                        join('src', 'npymath', 'npy_math_complex.c.src'),
                        join('src', 'npymath', 'halffloat.c')
                        ]
+    
+    compiler_type = customized_ccompiler().compiler_type
+    is_msvc = compiler_type == 'msvc'
+    is_msvc = is_msvc or (sys.platform == 'win32' and compiler_type in ('clang', 'intel'))
+
```

Comment:
I still think platform may be the way to go, but not `sys.platform`, rather the module.
```
import platform
platform.system() == 'Windows'
```